### PR TITLE
mesh : remove bug for current deployment

### PR DESCRIPTION
**Pull request by jeffj6123**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `./scripts/ci/test_static.sh` locally? (`pip install pylint flake8` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/mesh/azext_mesh/custom.py [hunk 1]
```diff
@@ -372,7 +372,7 @@ def _deploy_arm_template_core(cli_ctx, resource_group_name,  # pylint: disable=t
             sleep(timestep)
             wait_time += timestep
 
-        parsed_template = smc.deployments.validate(resource_group_name, deployment_name, properties).properties.additional_properties['validatedResources']
+        parsed_template = validation.properties.additional_properties['validatedResources']
```
Comment (williexu) [inline src/mesh/azext_mesh/custom.py]: Can you explain why this change is needed?

Comment (jeffj6123) [reply src/mesh/azext_mesh/custom.py]: Yes, so it appears there has been a change to the validate arm deployment endpoint which will now return an error about previous deployments in progress. The issue was this call could happen during a deployment, and thus return an error about a deployment in progress. Also since the template already gets validated before the deployment starts, this second check wasn't even necessary.
